packages/nextjs/app/start/_components/StartChallengesButton.tsx

```diff
@@ -3,14 +3,16 @@
 import { useCallback } from "react";
 import Link from "next/link";
 import { usePlausible } from "next-plausible";
+import { useLocalStorage } from "usehooks-ts";
 
 export const StartChallengesButton = () => {
   const plausible = usePlausible();
+  const [storedReferrer] = useLocalStorage<string | null>("originalReferrer", null);
 
   // TODO: test this later
   const handleCtaClick = useCallback(() => {
-    plausible("cta_from_start");
-  }, [plausible]);
+    plausible("cta_from_start", { props: { referrer: storedReferrer || "" } });
```

Comment:
I'm putting here the referrer inside the props. Not sure if that would work (on the backed side, it seems that we can pass it as referrer directly, not inside props)

Reply:
Was just researching on this didn't find any good issues or discussion but while chatting with gpt, it kind of made sense 

Plausible doesn't want you to spoof the user’s real referral data. So if you’re running in the browser, they assume document.referrer is honest. And that's why JS client (and hooks like usePlausible) are meant to be “just fire-and-forget” and hence things like `referrer`, `url` etc are auto captured

Reply:
Got it thankss!